classes/cyclonedx-export.bbclass

```diff
@@ -9,76 +9,133 @@ CVE_VERSION ??= "${PV}"
 
 CYCLONEDX_EXPORT_DIR ??= "${DEPLOY_DIR}/cyclonedx-export"
 CYCLONEDX_EXPORT_SBOM ??= "${CYCLONEDX_EXPORT_DIR}/bom.json"
+CYCLONEDX_EXPORT_VEX ??= "${CYCLONEDX_EXPORT_DIR}/vex.json"
 CYCLONEDX_EXPORT_TMP ??= "${TMPDIR}/cyclonedx-export"
 CYCLONEDX_EXPORT_LOCK ??= "${CYCLONEDX_EXPORT_TMP}/bom.lock"
 
 python do_cyclonedx_init() {
     import uuid
     from datetime import datetime
 
+    timestamp = datetime.now().astimezone().isoformat()
     sbom_dir = d.getVar("CYCLONEDX_EXPORT_DIR")
     bb.debug(2, "Creating cyclonedx directory: %s" % sbom_dir)
     bb.utils.mkdirhier(sbom_dir)
 
     bb.debug(2, "Creating empty sbom")
-    write_sbom(d, {
+    write_json(d.getVar("CYCLONEDX_EXPORT_SBOM"), {
         "bomFormat": "CycloneDX",
         "specVersion": "1.4",
         "serialNumber": "urn:uuid:" + str(uuid.uuid4()),
         "version": 1,
         "metadata": {
-            "timestamp": datetime.now().astimezone().isoformat(),
+            "timestamp": timestamp,
             "tools": [{"name": "yocto"}]
         },
         "components": []
     })
+
+    bb.debug(2, "Creating empty patched CVEs VEX file")
+    write_json(d.getVar("CYCLONEDX_EXPORT_VEX"), {
+        "bomFormat": "CycloneDX",
+        "specVersion": "1.4",
+        "serialNumber": "urn:uuid:" + str(uuid.uuid4()),
+        "version": 1,
+        "metadata": {
+            "timestamp": timestamp,
+            "tools": [{"name": "yocto"}]
+        },
+        "vulnerabilities": []
+    })
 }
 addhandler do_cyclonedx_init
 do_cyclonedx_init[eventmask] = "bb.event.BuildStarted"
 
 python do_cyclonedx_package_collect() {
+    import uuid
+    import oe.cve_check
+
     # load the bom
     name = d.getVar("CVE_PRODUCT")
     version = d.getVar("CVE_VERSION")
-    sbom = read_sbom(d)
+    sbom = read_json(d.getVar("CYCLONEDX_EXPORT_SBOM"))
+    vex = read_json(d.getVar("CYCLONEDX_EXPORT_VEX"))
 
     for pkg in generate_packages_list(name, version):
         if not next((c for c in sbom["components"] if c["cpe"] == pkg["cpe"]), None):
             sbom["components"].append(pkg)
-
+            bom_ref = pkg["bom-ref"]
+
+            # populate vex file with patched CVEs
+            for _, patched_cve in enumerate(oe.cve_check.get_patched_cves(d)):
+                bb.debug(2, f"Found patch for CVE {patched_cve} in {name}@{version}")
+                vex["vulnerabilities"].append({
+                    "id": patched_cve,
+                    # vex documents require a valid source, see https://github.com/DependencyTrack/dependency-track/issues/2977
+                    # this should always be NVD for yocto CVEs.
+                    "source": {"name": "NVD", "url": "https://nvd.nist.gov/"},
```

Comment:
if it's always CVE, then using `https://nvd.nist.gov/vuln/detail/CVE-IS-HERE` should be even better

Reply:
Ah yes, I actually wanted to do that but forgot, thanks :+1: